content/en/docs/tasks/observability/metrics/querying-metrics/test.sh

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC1090,SC2154,SC2155
+
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -u
+set -o pipefail
+
+source "tests/util/samples.sh"
+
+# @setup profile=demo
+kubectl label namespace default istio-injection=enabled --overwrite
+
+# Install Bookinfo sample
+startup_bookinfo_sample
+
+# Install Prometheus
+kubectl apply -f samples/addons/prometheus.yaml -n istio-system
+_wait_for_deployment istio-system prometheus
+
+# Fire a couple of requests
+_set_ingress_environment_variables
+export INGRESS_URL="$INGRESS_HOST:$INGRESS_PORT"
+echo "$INGRESS_URL"
+for _ in {1..50}; do
+    curl -s http://"$INGRESS_URL"/productpage > /dev/null
+done
+
+# Now check Prometheus dashboard for the metric. It should be present
+function urlencode() {
+    local value=$1
+    python3 -c "import urllib.parse; print(urllib.parse.quote('''$value'''))"
+}
+
+function query_prometheus() {
+    local query_expr=$1
+    local prometheus_api_root='localhost:9090/api/v1'
+    local encoded_query=$(urlencode "$query_expr")
+    curl -sg -m 3.0 http://"$prometheus_api_root"/query?query="$encoded_query"
+}
+
+function query_total_requests() {
+    query_prometheus 'istio_requests_total' | jq .data.result[0].metric.__name__
+}
+
+function query_requests_to_productpage() {
+    local query='istio_requests_total{destination_service="productpage.default.svc.cluster.local"}'
+    query_prometheus "$query" | jq .data.result[0].metric.destination_service
```

Comment:
Why aren't you using the generated snips for this and the other queries?
```suggestion
    query_prometheus "$snip_querying_istio_metrics_5" | jq '.data.result[0].metric.destination_service,.data.result[0].metric.destination_version'
```